r2dbc/boot-r2dbc-json-column/src/main/java/com/example/learning/config/MetricsExecutionListener.java

```diff
@@ -0,0 +1,124 @@
+package com.example.learning.config;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import io.r2dbc.proxy.core.MethodExecutionInfo;
+import io.r2dbc.proxy.core.QueryExecutionInfo;
+import io.r2dbc.proxy.listener.ProxyMethodExecutionListener;
+import io.r2dbc.proxy.support.QueryExecutionInfoFormatter;
+import java.time.Duration;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MetricsExecutionListener implements ProxyMethodExecutionListener {
```

Comment:
@coderabbitai, please help me to write junit to check the metrics in grafana container.